DIPs/DIP1013.md

```diff
@@ -32,102 +33,182 @@ who's writing the pull request. Standardizing the process makes sure that
-Fixing such issues MUST include a deprecation period for the current behavior,
-and an introduction of the new behavior as the default only at the end of the
+Under normal circumstances, the removal of a language fetaure should be
+accompanied with a deprecation period. "Language features" includes bugs in the
+current behavior upon which existing code depends, e.g. [Issue
+10378](https://issues.dlang.org/show_bug.cgi?id=10378). Fixing such issues
+should include a deprecation period for the current behavior and an
+introduction of the new behavior as the default only at the end of the period.
+If the D leadership determines that a change is critical enough to be
+implemented immediately, then of course, it can be implemented immediately, but
+in general, breaking changes should involve a deprecation period in order to
+avoid causing serious problems for existing D projects or giving the impression
+that D is unstable. Whether an "accepts-invalid" bug should include a
+deprecation period depends on the nature of the bug and the impact of the
+change, but in general, they should be fixed immediately without a deprecation
+period.
+
+Deprecations to language features must also update the [language deprecations
+page](https://dlang.org/deprecate.html) on dlang.org simultaneously. The
+deprecation message given by the compiler should contain the planned removal
+period and/or a reference to more information pertaining to the deprecation.
+
+Warnings should not be used in the deprecation process. Warnings are set as
+errors in many build systems (including dub), and would therefore prematurely
+break many user's code. The exception is when a change deprecates a feature
+which is intended to turn something into a warning. In this case, the code
+which would trigger the warning must also first go through a deprecation
 period.
 
-Deprecations to language features MUST also update the [language deprecations page](https://dlang.org/deprecate.html) on dlang.org simultaneously. The deprecation
-message given by the compiler SHOULD contain the planned removal period and/or a
-reference to more information pertaining to the deprecation.
+## Summary of the Steps in the Normal Deprecation Process
+
+### Steps at the time of deprecation
+
+1. Remove all uses of the to-be-deprecated feature/symbol from dmd, druntime,
+   and phobos (unless it's in something that is also being deprecated).
+2. Update the documentation to indicate in which release the feature/symbol will
+   be fully removed as well as give any additional, appropriate information about the
+   deprecation (e.g. what should be used instead). In the case of language
+   changes, this includes updating the [language deprecations
+   page](https://dlang.org/deprecate.html).
+3. Create a changelog entry.
+4. Deprecate the feature/symbol such that a deprecation message is printed with
+   the release that the feature/symbol will be removed (and thus will be affected by the
```

Comment:
this sentence makes no sense.  Maybe

    Deprecate the feature/symbol so that a deprecation message gets printed from the next release